### ISSUE TITLE
NGINX rules deviate from upstream defaults

### ISSUE THREAD

**Issue by janmashat** [wodby/nginx#93]
The [comment](https://github.com/wodby/nginx/blob/master/templates/presets/drupal9.conf.tmpl#L102) where this [change](https://github.com/wodby/nginx/pull/86) propagates states:

# Replica of regex from Drupals core .htaccess.
However I wasn't able to find a reference to these new file extensions at the source: https://git.drupalcode.org/project/drupal/-/blob/11.x/.htaccess

Now this has become a breaking change on our project where we serve legitimate .md files sitewide.

**Comment by mxr576** [comment]
In addition, the requested "security hardenings" in [#86] could be achieved by excluding those files from the scaffolded ones by Drupal Scaffold plugin: https://www.drupal.org/docs/develop/using-composer/using-drupals-composer-scaffold#toc_6

So the requested hardenings does not have to implemented in the web server level.

To be clear, the only concerning part for us is blocking markdown files to served globally, when there are legitimate use cases for that --- and to be fair/IMO, the README.md of Drupal core does not tell anything about the installed version, etc. I may discloses that Drupal is the app framework, but that could be guessed in many different ways.

(cc @elaman )

**Comment by elaman** [comment]
Fair point regarding .md files not being a direct vulnerability, although I think that publicly facing Markdown documents should be served out of public file system (eg `sites/default/files`.

Given that MD files are more often included with code to explain said code (modules, themes, libraries, internal docs, etc), we don't want website visitors to be able to gain access to information about code.

**Comment by mxr576** [comment]
With the rise of LLMs, especially AI agents and bots, serving content in Markdown has become common and, in many cases, expected. Blocking all URLs that end with `.md` is no longer a safe default and causes real issues today.

Markdown is increasingly used as a first-class, user-facing format, not just an internal source file. This trend is accelerating, not fading.

Some relevant context:

* [https://dri.es/the-third-audience](https://dri.es/the-third-audience)
* [https://pronovix.com/articles/make-your-developer-portal-ready-ai-agents](https://pronovix.com/articles/make-your-developer-portal-ready-ai-agents)

Given this shift, blanket blocking of `.md` URLs should be reconsidered or at least made opt-in.
